scm/driver/github/content.go

```diff
@@ -55,20 +76,19 @@ type content struct {
 	Type    string `json:"type"`
 }
 
-type contentUpdate struct {
-	Sha     string `json:"sha"`
-	Message string `json:"message"`
-	HTMLURL string `json:"html_url"`
-	Author  struct {
-		Name  string    `json:"name"`
-		Email string    `json:"email"`
-		Date  time.Time `json:"date"`
-	} `json:"author"`
-	Committer struct {
-		Name  string    `json:"name"`
-		Email string    `json:"email"`
-		Date  time.Time `json:"date"`
-	} `json:"committer"`
+type contentCreateUpdate struct {
+	Branch    string
+	Message   string
+	Content   []byte
+	SHA       string
+	Author    CommitAuthor
+	Committer CommitAuthor
+}
+
+type CommitAuthor struct {
+	Date  *time.Time
+	Name  string
+	Email string
```

Comment:
missing `json` tags to ensure correct naming.  Also why was time changed to a pointer?  do you want to use `omitempty` instead?